### PR TITLE
Qid linking to non hh cases

### PR DESCRIPTION
**Pull request by NickGrantONS**
# Motivation and Context
SPG & CE cases shouldn't create child HI cases when we link unaddressed questionnaires.

# What has changed
Limited individual case creation to HH cases only.

# How to test?
Run the ATs.

# Links
Trello: https://trello.com/c/GJvtm94j

### CODE REVIEW

**Comment by NickGrantONS** [comment]
[PQ Linking for CEs and SPGs (8)](https://trello.com/c/GJvtm94j/531-pq-linking-for-ces-and-spgs-8)

File: src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java [hunk 1]
```diff
@@ -37,17 +37,18 @@ public void processQuestionnaireLinked(
...
+        caze = caseService.prepareIndividualResponseCaseFromParentCase(caze);
 
-      caseService.emitCaseCreatedEvent(caze);
-    } else {
-      caze = caseService.getCaseByCaseId(UUID.fromString(uac.getCaseId()));
+        caseService.emitCaseCreatedEvent(caze);
+      }
     }
 
+    // TODO: This is wrong for CEs and SPGs but there is another ticket which deals with fixing this
```
Comment (AdamHawtin) [inline src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java]: 👍 For keeping these things visible

File: src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java [hunk 2]
```diff
@@ -346,6 +346,110 @@ public void testGoodIndividualQuestionnaireLinked() throws Exception {
...
+    uac.setQuestionnaireId(expectedQuestionnaireId);
+    managementEvent.getPayload().setUac(uac);
+
+    // WHEN
+    // Send questionnaire linked message
+    sendMessageAndDoNotExpectInboundMessage(
+        questionnaireLinkedQueue, managementEvent, outboundCaseQueue);
+
+    // THEN
+    // Check database HI Case created as expected
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java]: We shouldn't be creating a `HI` case here. It looks like the code is actually checking the parent case ID so only the comment is incorrect?

File: src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java [hunk 3]
```diff
@@ -346,6 +346,110 @@ public void testGoodIndividualQuestionnaireLinked() throws Exception {
...
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java]: Same as above, I think this comment about a `HI` case is incorrect/outdated?

File: src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java [hunk 4]
```diff
@@ -346,6 +346,110 @@ public void testGoodIndividualQuestionnaireLinked() throws Exception {
...
+    Case actualHHCase = caseRepository.findByCaseId(TEST_CASE_ID).get();
+
+    // Check database that HI Case is linked to UacQidLink
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java]: Same as above, I think this comment about a `HI` case is incorrect/outdated?

File: src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java [hunk 5]
```diff
@@ -346,6 +346,110 @@ public void testGoodIndividualQuestionnaireLinked() throws Exception {
...
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java]: Same as above, I think this comment about a `HI` case is incorrect/outdated?

File: src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java [hunk 6]
```diff
@@ -346,6 +346,108 @@ public void testGoodIndividualQuestionnaireLinked() throws Exception {
...
+
+    // WHEN
+    // Send questionnaire linked message
+    sendMessageAndDoNotExpectInboundMessage(
+        questionnaireLinkedQueue, managementEvent, outboundCaseQueue);
+
+    // THEN
+    Case actualHHCase = caseRepository.findByCaseId(TEST_CASE_ID).get();
+
+    // Check database that HH Case is linked to UacQidLink
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java]: Should be a `CE` case not `HH`?

File: src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java [hunk 7]
```diff
@@ -346,6 +346,108 @@ public void testGoodIndividualQuestionnaireLinked() throws Exception {
...
```
Comment (AdamHawtin) [inline src/test/java/uk/gov/ons/census/casesvc/messaging/QuestionnaireLinkedReceiverIT.java]: Should be an `SPG` case not `HH`?

**Pull request merged**